scripts/cluster_buildings.py

```diff
@@ -2,152 +2,144 @@
-    new_data = {"type": "FeatureCollection", "features": []}
-    for feature in data["features"]:
-        if feature["geometry"]["type"] == "Point":
-            new_feature = {
-                "type": "Feature",
-                "properties": {"id": feature["properties"]["id"]},
-                "geometry": {
-                    "type": feature["geometry"]["type"],
-                    "coordinates": feature["geometry"]["coordinates"],
-                },
-            }
-            new_data["features"].append(new_feature)
-
-    # Write new GeoJSON data to file
-    with open(output_file, "w") as f:
-        json.dump(new_data, f)
-
-
-def get_central_points_geojson(input_filepath, output_filepath, n_clusters):
-    # Load GeoJSON data from file
-    with open(input_filepath, "r") as f:
-        data = json.load(f)
 
-    # Extract coordinates of all points
-    points = []
-    for feature in data["features"]:
-        if feature["geometry"]["type"] == "Point":
-            points.append(feature["geometry"]["coordinates"])
-
-    # Convert points to NumPy array for use with scikit-learn
-    points = np.array(points)
-
-    # Perform k-means clustering
-    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(points)
-
-    # Compute centroids of each cluster
+def buildings_classification(input_file, crs, house_area_limit):
+    microgrid_buildings = gpd.read_file(input_file)
+    microgrid_buildings.rename(columns={"tags.building": "tags_building"}, inplace=True)
+    features = []
+    for row in microgrid_buildings.itertuples():
+        if row.geometry.type == "Polygon":
+            features.append(
+                {
+                    "properties": {
+                        "id": row.id,
+                        "tags_building": row.tags_building,
+                    },
+                    "geometry": row.geometry,
+                }
+            )
+    buildings_geodataframe = gpd.GeoDataFrame.from_features(features)
+    microgrid_buildings = microgrid_buildings.to_crs(crs)
+    area_m2 = microgrid_buildings.geometry.area
+    area_m2 = pd.Series(area_m2)
+    buildings_geodataframe["area_m2"] = area_m2
+    for index, row in buildings_geodataframe.iterrows():
+        if row.tags_building == "yes":
+            if row.area_m2 < house_area_limit:
+                buildings_geodataframe.loc[index, "tags_building"] = "house"
+            else:
+                buildings_geodataframe.loc[index, "tags_building"] = "yes"
+    return buildings_geodataframe
+    # buildings_geodataframe.to_file(output_file)
+
+
+def get_central_points_geojson(
+    input_filepath, output_filepath, n_clusters, crs, house_area_limit
+):
+    microgrid_buildings = buildings_classification(
+        input_filepath, crs, house_area_limit
+    )
+    # microgrid_buildings = gpd.read_file(input_filepath)
+    centroids_building = [
+        (row.geometry.centroid.x, row.geometry.centroid.y)
+        for row in microgrid_buildings.itertuples()
+    ]
```

Comment:
Not sure if the for loop is needed. What about `centroids_building = microgrid_buildings.geometry.centroid` ?

Reply:
I had tried not extracting individual coordinates, and directly using centroids_building = microgrid_buildings.geometry.centroid, but the kmeans "fit" method does not accept "point" elements

Reply:
Ok! :)